pallets/capacity/src/tests/rewards_provider_tests.rs

```diff
@@ -0,0 +1,268 @@
+		}, // truncated result
+		TestCase {
+			total_staked: 1_000_000,
+			amount_staked: 1000,
+			expected_reward: 4,
+			reward_pool: 10_000,
+		}, // hits the cap starting with this example
+		TestCase {
+			total_staked: 1_000_000,
+			amount_staked: 11000,
+			expected_reward: 42,
+			reward_pool: 10_000,
+		}, // > 0.0038% of total, reward = 11000*.0038
+		TestCase {
+			total_staked: 20_000_000,
+			amount_staked: 888_889,
+			expected_reward: 3378,
+			reward_pool: 11_000_000,
+		}, //  testnet/mainnet values
+	];
+	for tc in test_cases {
+		assert_eq!(
+			Capacity::era_staking_reward(tc.amount_staked, tc.total_staked, tc.reward_pool),
+			tc.expected_reward
+		);
+	}
+}
+
+#[test]
+fn list_unclaimed_rewards_returns_empty_set_when_no_staking() {
+	new_test_ext().execute_with(|| {
+		let account = 500u64;
+		let history: ProviderBoostHistory<Test> = ProviderBoostHistory::new();
+		ProviderBoostHistories::<Test>::set(account, Some(history));
+		let rewards = Capacity::list_unclaimed_rewards(&account).unwrap();
+		assert!(rewards.is_empty())
+	})
+}
+#[test]
+fn list_unclaimed_rewards_returns_empty_set_when_only_staked_this_era() {
+	new_test_ext().execute_with(|| {
+		system_run_to_block(5);
+		set_era_and_reward_pool(5u32, 1u32, 1000u64);
+		let account = 500u64;
+		let mut history: ProviderBoostHistory<Test> = ProviderBoostHistory::new();
+		history.add_era_balance(&5u32, &100u64);
+		ProviderBoostHistories::<Test>::set(account, Some(history));
+		let rewards = Capacity::list_unclaimed_rewards(&account).unwrap();
+		assert!(rewards.is_empty())
+	})
+}
+
+// Check that eligible amounts are only for what's staked an entire era.
+#[test]
+fn list_unclaimed_rewards_has_eligible_rewards() {
+	new_test_ext().execute_with(|| {
+		let account = 10_000u64;
+		let target: MessageSourceId = 10;
+		let amount = 1_000u64;
+
+		// staking 1k as of block 1, era 1 (1-10)
+		setup_provider(&account, &target, &amount, ProviderBoost);
+
+		// staking 2k as of block 11, era 2  (11-20)
+		run_to_block(11);
+		assert_ok!(Capacity::provider_boost(RuntimeOrigin::signed(account), target, amount));
+
+		//  staking 3k as of era 4, block 31, first block of era (31-40)
+		run_to_block(31);
+		assert_ok!(Capacity::provider_boost(RuntimeOrigin::signed(account), target, amount));
+
+		run_to_block(51);
+		assert_eq!(Capacity::get_current_era().era_index, 6u32);
+		assert_eq!(Capacity::get_current_era().started_at, 51u32);
+
+		// rewards for era 6 should not be returned; era 6 is current era and therefore ineligible.
+		// eligible amounts for rewards for eras should be:  1=0, 2=1k, 3=2k, 4=2k, 5=3k
+		let rewards = Capacity::list_unclaimed_rewards(&account).unwrap();
+		assert_eq!(rewards.len(), 5usize);
+		let expected_info: [UnclaimedRewardInfo<BalanceOf<Test>, BlockNumberFor<Test>>; 5] = [
+			UnclaimedRewardInfo {
+				reward_era: 1u32,
+				expires_at_block: 130,
+				staked_amount: 1000,
+				eligible_amount: 0,
+				earned_amount: 0,
+			},
```

Comment:
Should `UnclaimedRewardInfo` ever be able to return when `earned_amount` and/or `eligible_amount` is zero? Is that not a reward at all?